### ISSUE TITLE
Schema to EML

### ISSUE THREAD

**Issue by cboettig** [boettiger-lab/eml2schema#3]
@AlexLi0104 We'll also want to go from Schema.org into EML.  You've probably seen the minimal draft for this in https://github.com/boettiger-lab/eml2schema/blob/master/Notebooks/jq/schema_to_eml.jq .  I think this one will be easier, because in general Schema.org is a bit simpler than EML, so you will have fewer possible cases to work with.  Just wanted to mention this in case you are ever feeling stuck on the EML to Schema direction, working on the Schema to EML might be a good way to make some progress and get more familiar with the standards.  Eventually we want to be able to go in both directions anyway.

**Comment by AlexLi0104** [comment]
@cboettig 

Greetings! I just modified the schema_to_eml.jq file a bit and added the `creator` element. I re-knitted it in `jq_maps.md`. Please take a look at it at your convenience and see if that's right. I also renamed the `output` folder to `eml` and put it in `examples` folder, and deleted the original output folder.

Thank you very much!

**Comment by cboettig** [comment]
Great work so far.  Looks pretty good.  Note that `eml` doesn't ever have a `type` field, so you just want to drop the `"type": "Dataset"` part.  Instead, this would all go inside a `"dataset": {}` object.

`creator` mapping into schema looks good.

I think the `temporalCoverage` actually needs a `"calendarDate"` field before the actual dates (that was probably my mistake), but check against the other EML examples we now have.

Yup, definitely charge ahead on translating all the other fields in the example schema data into the corresponding EML structures; you should be able to find examples for how it should look in EML in the example set.  

nice work so far! ✨

**Comment by AlexLi0104** [comment]
@cboettig 

Hi! I just added the `calendarDate` to the `temporalCoverage,` and I also put the entire thing inside a `dataset` object.  Please let me know if there's anything else that I should pay attention to. I will continue to translate other fields. 

Thank you!

**Comment by cboettig** [comment]
Excellent. Next we will also want to put the `dataset` element itself inside an block of type `eml`, (see the examples in  the `emld` package). This should let us validate the EML (as shown in the README in `emld`), which is the best way to confirm the mapping is correct

**Comment by AlexLi0104** [comment]
@cboettig 

Hi! I just added a lot more elements to the jq script (based on the two Schema.org files that I have). However, I ran into the same problem that I described last time in the EML to Schema section: the objects in `creator` in `earthcube.json` are grouped in `[ ]`, but those in `schema-org-dataset.json` are not. Therefore, the script for the two files are not compatible (because I need to write `.creator[] | {...` in the one with `[ ]`, but  `.creator | {...` in the other one). I tried many ways but still can't fix this problem. Would you mind take a look at it and see if there's a good solution? There are also some other parts that I'm a bit confused, but I think this `[ ]` is the biggest issue in terms of compatibility.

Thank you!

**Comment by cboettig** [comment]
Yup, dealing with arguments that are only sometimes list elements is definitely going to be a challenge.  On the output side, it's technically okay if always put things inside `[]` whether there's only one or multiple elements in the list, but on the input side we probably need to do both.  

I suggest writing up a minimal example, like below, and and post it as a question to stackoverflow.

> How can one write a `jq` query that will extract a property that from an element that may appear as singleton or list?  e.g. extract the `url` property from the `creator` in both example JSON strings below: 

example 1:
```
{
"@type": "example1",
"creator":{
     "@type":"Organization",
     "url": "https://www.ncei.noaa.gov/"
     }
}
```
example2
```
{ 
  "@type": "example2",
  "creator": [{
    "@type":"Person",
    "url": "https://www.example.com/homepage"
    },
    {
    "url": "https://www.example.com/another"
     },
]
}
```

**Comment by cboettig** [comment]
@AlexLi0104 You might get better answers in SO, but conditional queries might be the way to go here:

```
jq(ex1, '.creator[].url? // .creator.url')
```

**Comment by AlexLi0104** [comment]
@cboettig 

Hi! I got an answer from StackOverflow and I think that problem is now solved! I have also made some additional changes to the script and I think it should work fine for the two Schema.org files that I have (`earthcube.json` and `schema-org-dataset`).  Please feel free to take a look since I'm not sure if each element is translated to the correct EML category, and there are still some elements in the files such as `variablesMeasured` that I can't find the corresponding category in EML. Also, I wonder if there are more Schema.org example files so that I could keep adding other elements. 

I am also trying to correct the [] issue on the eml-to-schema script, but it seems to be impacting other elements (probably because there are too many conditionals). I will keep trying to fix that and also keep adding other elements to that script.

Thank you very much!

**Comment by cboettig** [comment]
Sounds like very good progress, thanks for the update!  I'll take a look for some more schema.org examples and post them.  Keep me posted and I'll have some more detailed comments for you later this week.
